core/templates/pages/exploration-player-page/services/content-translation-manager.service.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to manage the content translations displayed.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { EventEmitter, Injectable } from '@angular/core';
+
+import { PlayerTranscriptService } from
+  'pages/exploration-player-page/services/player-transcript.service';
+import { StateCard } from
+  'domain/state_card/StateCardObjectFactory';
+import { WrittenTranslation } from
+  'domain/exploration/WrittenTranslationObjectFactory';
+import { SubtitledHtml } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { Schema } from 'services/schema-default-value.service';
+import { SchemaConstants } from
+  'components/forms/schema-based-editors/schema-constants';
+import INTERACTION_SPECS from 'pages/interaction-specs.constants.ajs';
+import { WrittenTranslations } from
+  'domain/exploration/WrittenTranslationsObjectFactory';
+import { SubtitledUnicode } from
+  'domain/exploration/SubtitledUnicodeObjectFactory';
+import { ExtensionTagAssemblerService } from
+  'services/extension-tag-assembler.service';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ContentTranslationManagerService {
+  _explorationLanguageCode: string;
+  _onStateCardContentUpdateEmitter: EventEmitter<void> = new EventEmitter();
+
+  constructor(
+    private playerTranscriptService: PlayerTranscriptService,
+    private extensionTagAssemblerService: ExtensionTagAssemblerService
+  ) {}
+
+  init(explorationLanguageCode: string) : void {
+    this._explorationLanguageCode = explorationLanguageCode;
+  }
+
+  get onStateCardContentUpdate(): EventEmitter<void> {
+    return this._onStateCardContentUpdateEmitter;
+  }
+
+  displayTranslations(languageCode: string) : void {
+    const cards = this.playerTranscriptService.transcript;
+
+    if (languageCode === this._explorationLanguageCode) {
+      this.playerTranscriptService.restoreImmutably();
+    } else {
+      cards.forEach(
+        card => this._displayTranslationsForCard(card, languageCode));
+    }
+
+    cards.forEach(card => {
+      // Because we do not have a mapping of the learner's input and Oppia's
```

Comment:
I'm not sure I understand this comment. Don't we know what state they fall under?